evm/deploy-contract.js

```diff
@@ -159,7 +157,7 @@ async function main(options) {
 
 const program = new Command();
 
-program.name('deploy-non-upgradable').description('Deploy non-upgradable contracts');
+program.name('deploy-contract').description('Deploy contracts using create3');
```

Comment:
can you add a flag (say `--deployer create3`) that decides which strategy to use (create3/create2/create)? The default can be create3, and we don't need a separate script for each approach I think. Extending this can also be a separate PR if you prefer

Reply:
That's an interesting option, although I don't think we really deploy any non-upgradable contracts using create2 or create other than the deployers themselves which have their own scripts right? Were you thinking of potentially merging those scripts into this one as well?

Reply:
We can discuss in more detail and include it in a separate PR, will resolve for the time being

Reply:
yep, don't need a separate script and gives the choice to use the appropriate method. There are some nicer properties of create2 over create3 that might make sense for certain deployments